### PR TITLE
v0.1.5

### PR DESCRIPTION
**Pull request by ausgerechnet**
- fixes performance issues
- ignores hypergeometric likelihood in standard settings due to performance issues

### CODE REVIEW

**Comment by martialblog** [comment]
@ausgerechnet This changes the API back to dataframes right?

File: association_measures/measures.py [hunk 1]
```diff
@@ -177,11 +172,11 @@ def calculate_measures(df, measures=None):
         if isinstance(measures[0], str):
             measures = [ams_all[k] for k in measures if k in ams_all.keys()]
     else:
-        measures = [ams_all[k] for k in ams_all if k != 'hypergeometric_likelihood']
```
Comment (martialblog) [inline association_measures/measures.py:180]: What's happening here?

File: tests/test_measures.py [hunk 2]
```diff
@@ -219,25 +240,33 @@ def test_log_likelihood_with_zeros(zero_dataframe):
...
 
-    df = sample_dataframe
-    df_ams = am.calculate_measures(df, ['hypergeometric_likelihood'])
-    assert df_ams['hypergeometric_likelihood'][0] == 5.776904234533874e-14
 
+# @pytest.mark.hypergeometric_likelihood
+# def test_hypergeometric_likelihood(sample_dataframe):
+#     df = sample_dataframe
+#     df_ams = am.calculate_measures(df, ['hypergeometric_likelihood'])
+#     assert df_ams['hypergeometric_likelihood'][0] == 5.776904234533874e-14
```
Comment (martialblog) [inline tests/test_measures.py:256]: Why are all the tests disabled?

**Comment by ausgerechnet** [comment]
yes, all functions accept dataframes again – I had thought that df.apply(…) would be equally as fast but apparently this is not the case. This is also why the hypergeometric likelihood ist so slow:
`c1 = df.apply(lambda row: choose(row['O11'] + row['O21'], row['O11']), axis=1)`
`c2 = df.apply(lambda row: choose(row['O12'] + row['O22'], row['O12']), axis=1)`
`c3 = df.apply(lambda row: choose(row['O11'] + row['O12'] + row['O21'] + row['O22'], row['O11'] + row['O12']), axis=1)`

File: tests/test_measures.py (hunk 2 above)
Comment (ausgerechnet) [reply tests/test_measures.py:256]: the hypergeometric likelihood is disabled in the standard interface `calculate_measures`. These tests thus don't work right now.

File: association_measures/measures.py (hunk 1 above)
Comment (ausgerechnet) [reply association_measures/measures.py:180]: When giving a list of `measures` to `calculate_measures`, the function translates those in the actual measures:
`for measure in measures: df[measure.__name__] = measure(df)`
The code before just determines what measures can be selected. Current implementation disables `hypergeometric_likelihood`.

**Comment by martialblog** [comment]
I know this is still a 0.* version, but the API should be somewhat stable at some point, since there might be people already using and depending on it.

File: tests/test_measures.py (hunk 2 above)
Comment (martialblog) [reply tests/test_measures.py:256]: I think a better option would be a skip mark and a comment. https://docs.pytest.org/en/stable/skipping.html

File: association_measures/measures.py (hunk 1 above)
Comment (martialblog) [reply association_measures/measures.py:180]: this seems a bit opaque, I'll have a look at it

**Comment by ausgerechnet** [comment]
the interface to `calculate_measures` didn't change, just the way the underlying functions operate.

and you're right, there *are* people already using it – me. the current version on PyPI however calculates wrong measures.

File: tests/test_measures.py (hunk 2 above)
Comment (ausgerechnet) [reply tests/test_measures.py]: this is very true. changed in latest commit.

**Pull request merged**